File: commonChars/JavaScript/commonChars.js
/*
 * Question:
 *  Write a program to read two names, 5 characters each, and output
 *  the common characters.
 *
 * Sample run:
 *  Enter a name: SALIH
 *  Enter a name: YASIN
 *  Common characters: S A I
 * */

function main(){
    let sUsrIn1, sUsrIn2, result=[];
    let i,j;

    console.log('Enter two names.\n5 character each.');
    while(true){
        sUsrIn1 = window.prompt('Enter name 1:').toLowerCase();
        sUsrIn2 = window.prompt('Enter name 2:').toLowerCase();

        if(sUsrIn1.length < 5 || sUsrIn2.length < 5){
            console.log('You should enter five characters exactly for each name!');
        }
        else{
            if(sUsrIn1.length > 5 || sUsrIn2.length > 5){
                console.log('Exceeded number of characters, only first five characters will be checked !!');
            }
            console.log('Common characters:\n');
            for(i=0; i<5; i++){
                for(j=0; j<5; j++){
                    if(sUsrIn1.charAt(i) === sUsrIn2.charAt(j)){
                        if(result.includes(sUsrIn2.charAt(j)) === false){
                            result.push(sUsrIn1.charAt(i));
                        }
                    }
                }
            }
            console.log(result);
            break;
        }
    }
}

main();

File: CMPE113,114,225,226/commonChars/C/commonChars.c
/*
 * Question:
 *  Write a program to read two names, 5 characters each, and output
 *  the common characters.
 *
 * Sample run:
 *  Enter a name: SALIH
 *  Enter a name: YASIN
 *  Common characters: S A I
 * */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define true 1
#define false 0

/*
 * NOTE:
 *  fgets will add '\0' character to the end of the string, remember that!
 * */

int main(void){
    char usr_inp1[10], usr_inp2[10];
    int i,j;

    // First input
    printf("Enter a name:\n");
    fgets(usr_inp1,10,stdin);

    // Second input
    printf("Enter a name:\n");
    fgets(usr_inp2,10,stdin);

    // Output
    printf("Common characters: ");
    for(i=0; i<5; i++){
        for(j=0; j<5; j++){
            if(usr_inp1[i] == usr_inp2[j]){
                printf("%c ",usr_inp1[i]);
            }
        }
    }
    return 0;
}